lib/godmin/engine_wrapper.rb

```diff
@@ -2,32 +2,47 @@ module Godmin
   class EngineWrapper
     attr_reader :engine
 
-    def initialize(obj)
-      @engine = find_engine(obj)
+    def initialize(controller)
+      @engine = find_engine(controller)
     end
 
-    def root
-      engine.root
+    def namespace
+      @namespace ||= engine.railtie_namespace
     end
 
-    def namespace
-      engine.railtie_namespace.to_s.underscore
+    def namespaced?
+      @namespaced ||= namespace.present?
+    end
+
+    def namespaced_path
+      @namespaced_path ||= begin
+        if namespaced?
+          namespace.name.classify.split("::").map(&:underscore)
+        else
+          []
+        end
+      end
+    end
+
+    def root
+      engine.root
     end
 
     private
 
-    def find_engine(obj)
-      mod = find_engine_module(obj)
-      if mod
-        "#{mod}::Engine".constantize
+    def find_engine(controller)
```

Comment:
No, but if we want to use `parents` it needs to include the introspection module from active support. Controllers do that and since we only use it for controllers currently I figured we could name it like that. If we need it for some other type of object later we should change it to something.. Not sure what though :P `introspectable_object`... :laughing: